run_tests.sh

```diff
@@ -9,160 +9,161 @@ export TF_CPP_MIN_LOG_LEVEL="3"
+                         "tests/defences/test_gaussian_noise.py" \
+                         "tests/defences/test_high_confidence.py" \
+                         "tests/defences/test_label_smoothing.py" \
+                         "tests/defences/test_neural_cleanse.py" \
+                         "tests/defences/test_pixel_defend.py" \
+                         "tests/defences/test_reverse_sigmoid.py" \
+                         "tests/defences/test_rounded.py" \
+                         "tests/defences/test_thermometer_encoding.py" \
+                         "tests/defences/test_variance_minimization.py" \
+                         "tests/defences/detector/evasion/subsetscanning/test_detector.py" \
+                         "tests/defences/detector/evasion/test_detector.py" \
+                         "tests/defences/detector/poison/test_activation_defence.py" \
+                         "tests/defences/detector/poison/test_clustering_analyzer.py" \
+                         "tests/defences/detector/poison/test_ground_truth_evaluator.py" \
+                         "tests/defences/detector/poison/test_provenance_defence.py" \
+                         "tests/defences/detector/poison/test_roni.py" \
+                         "tests/defences/detector/poison/test_spectral_signature_defense.py" )
+
+    declare -a metrics=("tests/metrics/test_gradient_check.py" \
+                        "tests/metrics/test_metrics.py" \
+                        "tests/metrics/test_verification_decision_trees.py" )
+
+    declare -a wrappers=("tests/wrappers/test_expectation.py" \
+                         "tests/wrappers/test_query_efficient_bb.py" \
+                         "tests/wrappers/test_wrapper.py" )
+
+    declare -a art=("tests/test_data_generators.py" \
+                    "tests/test_utils.py" \
+                    "tests/test_visualization.py" )
+
+    tests_modules=("attacks" \
+                   "classifiers" \
+                   "object_detectors" \
+                   "speech_recognizers" \
+                   "defences" \
+                   "metrics" \
+                   "wrappers" \
+                   "art" )
+
+    # --------------------------------------------------------------------------------------------------- CODE TO RUN TESTS
+
+    run_test () {
+      test=$1
+      test_file_name="$(echo ${test} | rev | cut -d'/' -f1 | rev)"
+
+      echo $'\n\n'
+      echo "######################################################################"
+      echo ${test}
+      echo "######################################################################"
+      coverage run --append -m unittest -v ${test}
+      if [[ $? -ne 0 ]]; then exit_code=1; echo "Failed $test"; fi
+    }
+
+    for tests_module in "${tests_modules[@]}"; do
+      tests="$tests_module[@]"
+      for test in "${!tests}"; do
+         run_test ${test}
+      done
+    done
+
+    #bash <(curl -s https://codecov.io/bash)
+
+
+    exit ${exit_code}
```

Comment:
line 168 and 165 should also be outside of this else statement. This will be solved if the `if/else` statement is removed as mentioned in the previous comment above.